### ISSUE TITLE
Make Debug log in Builds more usable

### ISSUE THREAD

**Issue by krille90** [unitystation/unitystation#1472]
## Description
Currently the Debug log in builds (press F5) isn't scrollable, which makes not so nice to work with.
This should be improved for better debugging.

**Comment by fomalsd** [comment]
also colour coding would be nice to have (Trace = light gray, Info = white, Warning = yellow, Error = red)

**Comment by JaytleBee** [comment]
Writing the log to the disk would be killer (for crashes and cross-referencing, etc.)! Also, I hate to nitpick but isn't this more a feature than a bug?

**Comment by christmoore** [comment]
I would recommend rather than using the in-house console that we utilize one of the unity assets that are available for this. The more popular one from looking at a few stackoverflow posts is the [Unity Log Viewer](https://github.com/aliessmael/Unity-Logs-Viewer/). Version 1.7 compiles correctly for the current project, and has a lot more utility than the one we currently have.

Basically the only change we'll have to do is add the latest 1.7 files to our UnityProject Assets folder. Once that is done, there are two simple things we have to do:
1. Create the Reporter Manager using the new taskbar Reporter drop-down menu:
  ![](https://i.imgur.com/dRB5Yt9.gif)
  
  
2. Ensure that the Reporter.cs script is at the top of the Script Execution Order of the project
  ![](https://i.imgur.com/B0QQ7sM.gif)

Basically, a pull request that implements this will:  
Add the files relevant to the asset  
Create the Reporter Manager once in the starting scene where we build - which is the lobby scene.  
  
It is unlikely that anything will ever take precedent over the Reporter.cs's default script order of -12000 so nothing needs to be done there. Maybe just a short wiki post detailing that we use this and to make sure it's at the top of the execution order.

---

To use this, all you have to do is hold down the left-mouse button and make a circular motion while a build is running, and the console menu will appear:
![](https://i.imgur.com/tne0X9s.gif)

Here are a few examples of its uses:

1. The console log allows you to click to see the context of the bug - can see the stack where the error occured.

2. Filter out warnings and informational messages.

3. Click to save logs (Funny that you just posted how that would be nice @JaytleBee)
![](https://i.imgur.com/NB0nO41.gif)
![](https://i.imgur.com/ThGYEAe.gif)

If this is approved you can assign the task to me and I will remove usage of our old console logger and implement this.

**Comment by DooblyNoobly** [comment]
@MrMepper not bad, it feels like a bit of overkill though. We just want something simple that takes up only 1/3 to 1/2 of the screen when active. I think what we have already is sufficient enough, all we need to do is add a scroll bar to the scrollrect and throw in some color tags to the different errors.

I've been laying out the console so that eventually we can add an input field at the bottom and execute commands from there also.

**Comment by christmoore** [comment]
I'll take a crack at this!

Also final note, I've also found [this debug console](https://github.com/yasirkula/UnityIngameDebugConsole) which I've tested as well. I feel it's a lot less overkill, and it also has the functionality requested, as well as a method for providing input already built into it.


What it does is it sits on the screen for the build as a notification window.

![image](https://user-images.githubusercontent.com/8658239/51813485-314e1280-227c-11e9-8ed8-255c270490c1.png)

Which you can then click and it will cause the debugger to expand:

![image](https://user-images.githubusercontent.com/8658239/51813536-7ffbac80-227c-11e9-9e31-a80ad8e264c5.png)

The repo has documentation in the readme for creating custom commands. If this one is a no-go I will continue with our current console.

**Comment by DooblyNoobly** [comment]
I like that version much better. Looks like it is MIT as well. Go ahead and replace old one with that. Make sure that it is only visible when you press f5 and it can also be hidden by pressing f5 again.

**Comment by christmoore** [comment]
Great! Assign it over and I'll begin the changes tomorrow.
